src/routing/dispatcher.rs

```diff
@@ -100,6 +104,28 @@ impl Dispatcher {
                 message,
                 dest_info,
             } => {
+                if let (Variant::JoinRequest(join_request), Some(sender)) =
```

Comment:
you can have this block of code inside the `handle_join_request` function, 
so that the handling logic can be clustered ?

Reply:
The issue I faced there was we don't have access to the `comm` object to call `is_reachable()`

Reply:
Ah, right, it's in the separate object.
ok then